IdentityCore/src/telemetry/request_telemetry/MSIDLastRequestTelemetrySerializedItem.m

```diff
@@ -92,19 +99,39 @@ - (NSString *)serializeErrorsInfoWithCurrentStringSize:(NSUInteger)startLength
             // Only add next error into string if the resulting string smaller than 4KB, otherwise stop building
             // the string
             if ([newFailedRequestsString lengthOfBytesUsingEncoding:NSUTF8StringEncoding] +
-                    [newErrorMessagesString lengthOfBytesUsingEncoding:NSUTF8StringEncoding] + startLength < 4000)
+                    [newErrorMessagesString lengthOfBytesUsingEncoding:NSUTF8StringEncoding] + startLength < [MSIDCurrentRequestTelemetrySerializedItem telemetryStringSizeLimit])
             {
                 failedRequestsString = newFailedRequestsString;
                 errorMessagesString = newErrorMessagesString;
             }
             else
             {
-                break;
+                [self addRemainingErrorsToUnserializedTelemetry:i];
+                
+                return [NSString stringWithFormat:@"%@|%@", failedRequestsString, errorMessagesString];
             }
         }
     }
     NSString *telemetryString = [NSString stringWithFormat:@"%@|%@", failedRequestsString, errorMessagesString];
     return telemetryString;
 }
 
+- (NSMutableArray<MSIDRequestTelemetryErrorInfo *> *)getUnserializedTelemetry
+{
+    return self.unserializedErrors;
+}
+
+- (void)addRemainingErrorsToUnserializedTelemetry:(int)index
+{
+    if (!self.unserializedErrors)
+    {
+        self.unserializedErrors = [NSMutableArray<MSIDRequestTelemetryErrorInfo *> new];
+    }
+    
+    for (int i = 0; i <= index; i++)
+    {
+        [self.unserializedErrors addObject:self.errorsInfo[i]];
```

Comment:
same question here - any concurrency issues?

Reply:
I don't think so here either. In MSIDLastRequestTelemetry [self.telemetrySerializedItem serialize] is called from a dispatch_sync block, but [self.telemetrySerializedItem getUnserializedTelemetry] only gets called from a dispatch_barrier block, so it will wait until all queued up serialize operations are finished.

It does seem like changing MSIDLastRequestTelemetrySerializedItem from a single-use object instance to a property causes the thread protection here to be backwards, where the read operation is protected instead of the write..